core/appmodule/event.go

```diff
@@ -0,0 +1,37 @@
+package appmodule
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/runtime/protoiface"
+)
+
+// HasEventListeners is the extension interface that modules should implement to register
+// event listeners.
+type HasEventListeners interface {
+	AppModule
+
+	// RegisterEventListeners registers the module's events listeners.
+	RegisterEventListeners(registrar *EventListenerRegistrar)
+}
```

Comment:
I have a question on why to expose `EventListenerRegistrar` in core, shouldn't that be runtime?

The core interface could simply be:

```suggestion
type HasEventListeners interface {
	AppModule

	RegisterEventListeners() []func(context.Context, protoiface.MessageV1) 
	RegisterEventInterceptors() []func(context.Context, protoiface.MessageV1) error
}
```

Reply:
It's just a more ergonomic API this way I think

Reply:
EventListenerRegistrar is just a dummy struct to make using generics easy